sklearn/_callbacks.py

```diff
@@ -0,0 +1,58 @@
+# License: BSD 3 clause
+from typing import List, Callable
+from abc import ABC, abstractmethod
+
+import numpy as np
+
+CALLBACK_PARAM_TYPES = {
+    "n_iter": int,
+    "max_iter": int,
+    "loss": (float, dict),
+    "score": (float, dict),
+    "validation_loss": (float, dict),
+    "validation_score": (float, dict),
+    "coef": np.ndarray,
+    "intercept": (np.ndarray, float)
+}
+
+
+def _check_callback_params(**kwargs):
```

Comment:
Shouldn't we let each callback independently validate its data?

My question might not make sense but I don't see this being used anywhere except in the tests